src/GoogleAnalyticsTracker.js

```diff
@@ -0,0 +1,199 @@
+   */
+  trackEvent(category, action, optionalValues = {}) {
+    GoogleAnalyticsBridge.trackEvent(this.id, ...arguments);
+  }
+
+  /**
+   * Track the current screen/view with custom dimension values
+   * @param  {String} screenName The name of the current screen
+   * @param  {Object} customDimensionValues An object containing custom dimension key/value pairs
+   */
+  trackScreenViewWithCustomDimensionValues(screenName, customDimensionValues) {
+    const formattedCustomDimensions = this.transformCustomDimensionsFieldsToIndexes(customDimensionValues);
+    GoogleAnalyticsBridge.trackScreenViewWithCustomDimensionValues(this.id, screenName, formattedCustomDimensions);
+  }
+
+  /**
+   * Track an event that has occured with custom dimension values
+   * @param  {String} category       The event category
+   * @param  {String} action         The event action
+   * @param  {Object} optionalValues An object containing optional label and value
+   * @param  {Object} customDimensionValues An object containing custom dimension key/value pairs
+   */
+  trackEventWithCustomDimensionValues(category, action, optionalValues = {}, customDimensionValues) {
+    const formattedCustomDimensions = this.transformCustomDimensionsFieldsToIndexes(customDimensionValues);
+    GoogleAnalyticsBridge.trackEventWithCustomDimensionValues(this.id, category, action, optionalValues, formattedCustomDimensions);
+  }
+
+  /**
+   * Track an event that has occured
+   * @param  {String} category       The event category
+   * @param  {Number} value         	The timing measurement in milliseconds
+   * @param  {Object} optionalValues An object containing optional name and label
+   */
+  trackTiming(category, value, optionalValues = {}) {
+    GoogleAnalyticsBridge.trackTiming(this.id, ...arguments);
+  }
+
+  /**
+   * Track a purchase event. This uses the Enhanced Ecommerce GA feature.
+   * @param  {Object} product       An object with product values
+   * @param  {Object} transaction   An object with transaction values
+   * @param  {String} eventCategory The event category, defaults to Ecommerce
+   * @param  {String} eventAction   The event action, defaults to Purchase
+   */
+  trackPurchaseEvent(product = {}, transaction = {}, eventCategory = "Ecommerce", eventAction = "Purchase") {
+    GoogleAnalyticsBridge.trackPurchaseEvent(this.id, ...arguments);
+  }
+
+  /**
+   * Track a purchase event. This uses the Enhanced Ecommerce GA feature.
+   * @param  {Array} products       An array with products
+   * @param  {Object} transaction   An object with transaction values
+   * @param  {String} eventCategory The event category, defaults to Ecommerce
+   * @param  {String} eventAction   The event action, defaults to Purchase
+   */
+  trackMultiProductsPurchaseEvent(products = [], transaction = {}, eventCategory = "Ecommerce", eventAction = "Purchase") {
+    GoogleAnalyticsBridge.trackMultiProductsPurchaseEvent(this.id, ...arguments);
+  }
+
```

Comment:
[trackMultiProductsPurchaseEventWithCustomDimensionValues](https://github.com/idehub/react-native-google-analytics-bridge/blob/master/index.js#L94) also lacking from the implementation here.